src/main.ts

```diff
@@ -22,6 +24,21 @@ export default class ShikiPlugin extends Plugin {
 		this.loadedSettings = structuredClone(this.settings);
 		this.addSettingTab(new ShikiSettingsTab(this));
 
+		this.updateTheme();
+		this.themeObserver = new MutationObserver(mutations => {
```

Comment:
Using [this](<https://github.com/obsidianmd/obsidian-api/blob/103ff76a0712a02dd0da28646b5a6b0ba6686d66/obsidian.d.ts#L4988>) seems like a better approach than a mutation observer.